language/move-prover/bytecode/src/options.rs

```diff
@@ -29,6 +29,45 @@ impl AutoTraceLevel {
     }
 }
 
+/// Describes native method implementing mutable borrow semantics for a given type in an alternative
+/// storage model (returning &mut without taking appropriate &mut as a parameter, much like
+/// vector::borrow_mut)
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BorrowNative {
+    /// Method's module address
+    pub mod_addr: String,
+    /// Method's module name
+    pub mod_name: String,
+    /// Method's name
+    pub name: String,
+    /// Name of a read operation for a given type (in Boogie)
+    pub read_op: String,
+    /// Name of a Write operation for a given type (in Boogie)
+    pub write_op: String,
+    /// Method's type parameter index describing a give type
+    pub tparam_idx: usize,
```

Comment:
This seems to assume that the borrow type is one of the type parameters? What if the function has no type parameter or if we have type params `<K, V>` but the borrow type is `Wrapper<V>`? 
I'm fine with the way it is currently too if you think allowing those scenarios would be overgeneralization.

Reply:
I agree with your assessment of the limitation, but I am not sure how much it makes sense to generalize it at this point. I'd think that the current solution is already more general than the previous one which assumed that the required type is a type parameter of the argument (e.g., `Table<K,V>`) and it covers all the cases we need to cover right now.

If someone has a suggestion on how/why to generalize it further I am all ears, but perhaps we should consider it only if further need arises?

Reply:
After pondering this for a bit longer I realized that we probably don't need the type info at all as it was used only to figure out which "access" functions to use in the bytecode translator.